### PR TITLE
Petri Net to Lens

### PR DESCRIPTION
**Pull request by kris-brown**
`Trms`? In my `SymExprs`? 

Additionally, Labelled Petri Net to Lens converter.

### CODE REVIEW

**Comment by jpfairbanks** [comment]
Can we rename `to_lens` to `symbolic_dynamics` or `lens_dynamics` or something less OOP?

File: src/stdlib/models/Petri.jl [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+using ...StdTheories: ThRing
+using ....Syntax
+using ....Util
+using ....Dsl
+
+CtxTrm(i) = Trm(Lvl(i;context=true)) # upstream this?
+
+function lens_dynamics(p::LabelledPetriNet, T::Type=ThRing)
+  i_Ob = lookup(gettheory(T), Default())
+  pos = Context([(Name(s),Typ(i_Ob)) for s in p[:sname]])
```
Comment (olynch) [inline src/stdlib/models/Petri.jl:17]: Make an overload for `Context` that takes in a list of names and gives them all the default type.

File: src/stdlib/models/Petri.jl [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  dom_dir = Context([(Name(Symbol("d$s")),Typ(i_Ob)) for s in p[:sname]])
+  codom_dir = Context([(Name(t),Typ(i_Ob)) for t in p[:tname]])
+  dom = SimpleArena{T}(pos, dom_dir)
+  codom = SimpleArena{T}(pos, codom_dir)
+  expose = CtxTrm.(1:nparts(p,:S))
```
Comment (olynch) [inline src/stdlib/models/Petri.jl:22]: You could do this with `ContextMaps.Impl.id(pos)`

File: src/stdlib/models/Petri.jl [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  # Compute update, starting with each reaction rate
+  rxn_rates = map(parts(p, :T)) do t
+    tlvl = CtxTrm(t+nparts(p, :S))
+    i = CtxTrm.(p[incident(p, t, :it),:is])
+    sumterm = foldl((a,b)-> @term(T,$a * $b), i; init=@term(T, one()))
```
Comment (olynch) [inline src/stdlib/models/Petri.jl:27]: You can just use `one`

File: test/stdlib/Lenses.jl [hunk 4]
```diff
@@ -26,6 +27,10 @@ sir = @lens ThRing begin
   end
 end
 
+sir_pn = LabelledPetriNet([:s, :i, :r], 
+                          :β => ((:s, :i), (:i, :i)), :γ => (:i, :r))
+sir2 = lens_dynamics(sir_pn) # equivalent up to ring axioms
```
Comment (olynch) [inline test/stdlib/Lenses.jl:32]: I want one `@test`

**Review by olynch — CHANGES_REQUESTED** [review]

**Comment by jpfairbanks** [comment]
I resolved the merge conflicts in the web GUI. Now we can't rebase and merge commits and squashing are disabled for this repo.

**Comment by olynch** [comment]
> I resolved the merge conflicts in the web GUI. Now we can't rebase and merge commits and squashing are disabled for this repo.

Yeah, idk what the github "resolve merge conflicts action" actually does; I rebased this on top of `mtk` using magit and hopefully it should work now.

**Comment by olynch** [comment]
The windows failure is a CI failure, not a code failure...

**Pull request merged**